### PR TITLE
Support secondary public IP addresses for virtual guest

### PR DESCRIPTION
**Pull request by minsikl**
- Added secondary_ip_count and secondary_ip_addresses arguments to virtual_guest.
- This PR will support functions for #65

### CODE REVIEW

File: docs/resources/softlayer_virtual_guest.md [hunk 1]
```diff
@@ -128,6 +128,10 @@ The following arguments are supported:
     * Provides a primary public IPv6 address.
     * *Optional*
     * *Default*: false
+*   `secondary_ip_count` | *int*
+    * Provides secondary public IPv4 addresses. Acceptable values are 4 and 8. 
+    * *Optional*
+    * *Default*: 0
```
Comment (renier) [inline docs/resources/softlayer_virtual_guest.md]: I think it is a little confusing saying that the default is 0 when acceptable values are 4 and 8. We should keep that to ourselves, and not say there is any default.

File: softlayer/resource_softlayer_virtual_guest.go [hunk 2]
```diff
@@ -270,6 +270,19 @@ func resourceSoftLayerVirtualGuest() *schema.Resource {
 				Computed: true,
 			},
 
+			"secondary_ip_count": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ForceNew: true,
+				Default:  0,
```
Comment (renier) [inline softlayer/resource_softlayer_virtual_guest.go]: int type doesn't need to specify default of 0 I believe. That is the type's default already.

File: softlayer/resource_softlayer_virtual_guest.go [hunk 3]
```diff
@@ -718,6 +765,29 @@ func resourceSoftLayerVirtualGuestRead(d *schema.ResourceData, meta interface{})
...
+	if result.PrimaryIpAddress != nil {
+		secondarySubnetResult, err := services.GetAccountService(meta.(ProviderConfig).SoftLayerSession()).
+			Mask("ipAddresses[id,ipAddress]").
+			Filter(filter.Build(filter.Path("publicSubnets.endPointIpAddress.ipAddress").Eq(*result.PrimaryIpAddress))).
+			GetPublicSubnets()
+		if err != nil {
+			log.Printf("Error getting secondary Ip addresses: %s", err)
+		}
+
+		secondaryIps := make([]string, 0)
```
Comment (renier) [inline softlayer/resource_softlayer_virtual_guest.go:778]: if we know the number of secondary ips ahead of time, we can set the capacity in this make statement. `make([]string, 0, len(secondarySubnetResult))`. It is a small optimization.

Comment (minsikl) [reply softlayer/resource_softlayer_virtual_guest.go:778]: To get the number of secondary ips, I need to execute for loop and count it as follows:
```
		numberOfSecondaryIps := 0
		for _, subnet := range secondarySubnetResult {
			numberOfSecondaryIps += len(subnet)
		}
		secondaryIps := make([]string, 0, numberOfSecondaryIps)
```
It's difficult to expect optimization.

File: softlayer/resource_softlayer_virtual_guest.go (hunk 2 above)
Comment (minsikl) [reply softlayer/resource_softlayer_virtual_guest.go]: done.

File: docs/resources/softlayer_virtual_guest.md (hunk 1 above)
Comment (minsikl) [reply docs/resources/softlayer_virtual_guest.md]: done.

File: softlayer/resource_softlayer_virtual_guest.go (hunk 3 above)
Comment (renier) [reply softlayer/resource_softlayer_virtual_guest.go:778]: ok, I see

**Pull request merged**